### ISSUE TITLE
MC Eternal 100% ram usage 

### ISSUE THREAD

**Issue by Briig** [adam9899/Issues#628]
I am having trouble managing my server right now because every couple hours the server goes to 100% ram usage and i have 10 GB of ram so that should be plenty am I right.

**Comment by ghost** [comment]
Not really. It's recommended that you allocate 6GB ***MINIMUM***. If you have 10GB system memory, with 6gb allocated that leaves 4gb for everything else, including the OS. I personally use 10GB for MCEternal, on a 16GB system. You shouldn't be surprised if it goes to 100%, because 10GB is pretty low for a heavily modded server. I'm using 13/16GB right now with just MC and a browser running. You might want to close the issue.

**Comment by ogradyjd** [comment]
Actually, this might be a conflagration of other problems.  I just dealt with the Hats mod being a serious drain on resources.  I was using lag goggles to see why my 10GB server (that's 10GB allocated to MC Eternal on a system with 32GB of RAM) kept degrading after only a few hours or so, and found that Hats was being an asshole.   Profiling FPS showed the mod taking up 45%+ of the event subscribers and 2000+ ns/f for entities.   

As soon as I changed the randomMobHat value to 0, my performance got a ton better.   I think it's not just hats though, but some kind of weird interaction with `mod_lavacow:raven` entities.   With randomMobHat set to 5 (the default), I was seeing chunks with thousands of ravens in them if you sat in the same area long enough.  When I ran:
```
/kill @e[type=mod_lavacow:raven]
```
... I also saw an immediate increase in performance.  When I turned  `randomMobHat` to "0", the raven entity explosion problem went away and my performance stayed pretty stable for around 12 hours at a clip. I'm thinking of upping the memory to 12GB and running VisualVM or YourKit on the JVM to see if that's enough or too much, but that's if I can find the time.

Sorry to the MC Eternal devs.  I know in git issues I'm not supposed to be actually helpful.  Maybe I should redact all of the above and replace it with "It broke.  You fix?"  :)  Regardless, you might think of removing the Hats mod. It really brings nothing to the table and it seems to be problematic for everyone's performance.  Replace it with something like Reliquary, Draconic Evolution, Environmental Tech, or Woot.

**Comment by ogradyjd** [comment]
Oh, one other thing.  I keep getting growing pods of trolls around my base as well, which also absolutely kills performance. It looks like an Ice and Fire mod issue where they haven't handled the problem of someone sticking to their base for long stretches.  Trolls spawn, get caught out in the sun, turn to stone, and then sit there forever.   If there's some config to have the trolls stone forms degrade and die over a certain amount of time, you really want to set that.  Otherwise, not really an MC Eternal problem, but we have to deal with it anyway.

Again, I just run:

```
/kill @e[type=iceandfire:if_troll]
```
... once in a while and the performance gets immediately better. 
![2020-05-23_11 28 50](https://user-images.githubusercontent.com/4014049/82734696-a869ae80-9cea-11ea-9123-aecf270b0239.png)

**Comment by Briig** [comment]
yeah its a MC eternal server dedicated to just running high Mod Mod Packs rented from apex hosting with 10GB of memory i dont see that not being enough for 3 people 5 on uptime

**Comment by ogradyjd** [comment]
@Briig You have to remember that most of those hosting services are running something like Kubernetes or Docker or VMWare enterprise.  At best, these systems slice off "time" from the main RAM banks claiming that you have 10GB, where in reality you have only 8GB at any given point and because of the sharing subsystems, that RAM is actually slower than they advertise.  The only way around this is to pay for a true co-located server where you're the only person on it and there's no middleware adding a layer sucking up resources and slowing things down.

That's not to say the services aren't good, so long as their admins know how to tune them right.  The problem comes when they advertise more than they are "technically" giving you.

That said, MC Eternal is the biggest pack I've run since All The Mods 3.  I'm playing it on an Alienware 17 r4.  Without the Hats mod, it seems to be okay for a half day with 10GB.  I'm thinking in a true isolated server, 10GB would be more than enough for 5 or more players. 12GB for say 10 players.

**Comment by Mohl** [comment]
I'm running MC Eternal on a dedicated (self-hosted) 16 GB Linux machine and I allocate 10GB - for 2 players. I was already thinking of upgrading to 32GB, but I will try to verify @ogradyjd's suggestions.
